code/ui/PlayerInfo.cs

```diff
@@ -38,16 +37,72 @@ public override void Tick()
                 return;
             }
 
+            RoleLabel.Text = $"{player.Role.ToString()}";
+        }
+    }
+
+    public class IndicatorsPanel : Panel
+    {
+        public BarPanel HealthBar { set; get; }
+        public BarPanel AmmoBar { set; get; }
+
+        public bool invisibleAmmo { set; get; }
+
+        public IndicatorsPanel(Panel parent)
+        {
+            Parent = parent;
+
+            HealthBar = new BarPanel(this, "", "healthlabel");
+            HealthBar.AddClass("health");
+
+            AmmoBar = new BarPanel(this, "", "ammolabel");
+            AmmoBar.AddClass("ammo");
+        }
+
+        public override void Tick()
+        {
+            TTTPlayer player = Local.Pawn as TTTPlayer;
+
+            if (player == null)
+            {
+                return;
+            }
+
+            HealthBar.TextLabel.Text = $"{player.Health:n0}";
+
             var weapon = player.ActiveChild as Weapon;
 
             if (weapon != null)
             {
-                Weapon.Text = $"{weapon.AmmoClip}";
+                if (invisibleAmmo)
```

Comment:
Does `invisibleAmmo` mean that the current weapon has no ammo ? i.e melee weapon?

Would a name like `HasAmmo` be better? or something along the lines of that.

Reply:
Yea currently a dirty workaround as well as long as there isn't any Inventory System :D But yea, you are right, can be directly connected to the Weapon (anyways, should just trigger on ammo change of a weapon).

Reply:
Marking as resolved, as long as we just fix it in the future.